tests/components/august/test_config_flow.py

```diff
@@ -0,0 +1,195 @@
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.august.config_flow.AugustGateway.authenticate",
+        side_effect=CannotConnect,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_LOGIN_METHOD: "email",
+                CONF_USERNAME: "my@email.tld",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_needs_validate(hass):
+    """Test we present validation when we need to validate."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.august.config_flow.AugustGateway.authenticate",
+        side_effect=RequireValidation,
+    ), patch(
+        "homeassistant.components.august.gateway.Authenticator.send_verification_code",
+        return_value=True,
+    ) as mock_send_verification_code:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_LOGIN_METHOD: "email",
+                CONF_USERNAME: "my@email.tld",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+    assert len(mock_send_verification_code.mock_calls) == 1
+    assert result2["type"] == "form"
+    assert result2["errors"] is None
+    assert result2["step_id"] == "validation"
+
+    # Try with the WRONG verification code give us the form back again
+    with patch(
+        "homeassistant.components.august.config_flow.AugustGateway.authenticate",
+        side_effect=RequireValidation,
+    ), patch(
+        "homeassistant.components.august.gateway.Authenticator.validate_verification_code",
+        return_value=ValidationResult.INVALID_VERIFICATION_CODE,
+    ) as mock_validate_verification_code, patch(
+        "homeassistant.components.august.gateway.Authenticator.send_verification_code",
+        return_value=True,
+    ) as mock_send_verification_code, patch(
+        "homeassistant.components.august.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.august.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        result3 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {VERIFICATION_CODE_KEY: "incorrect"},
+        )
+
+    # Make sure we do not resend the code again
+    # so they have a chance to retry
+    assert len(mock_send_verification_code.mock_calls) == 0
+    assert len(mock_validate_verification_code.mock_calls) == 1
+    assert result3["type"] == "form"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5068548)